registration-processor/workflow-engine/registration-processor-workflow-manager-service/src/main/java/io/mosip/registration/processor/workflowmanager/verticle/UserNotificationApi.java

```diff
@@ -0,0 +1,316 @@
+				updateAudit(description, UserNotificationRequestDTO.getRid(), false, user);
+				throw new UserNotificationException(PlatformErrorMessages.RPR_UNA_RID_NOT_FOUND.getCode(),
+						String.format(PlatformErrorMessages.RPR_UNA_RID_NOT_FOUND.getMessage(), rid));
+			}
+			if (!registrationStatusDto.getRegistrationType().equals(UserNotificationRequestDTO.getRegType())) {
+				description.setMessage(PlatformErrorMessages.RPR_UNA_REGTYPE_NOT_MATCHING.getMessage());
+				description.setCode(PlatformErrorMessages.RPR_UNA_REGTYPE_NOT_MATCHING.getCode());
+				updateAudit(description, UserNotificationRequestDTO.getRid(), false, user);
+				throw new UserNotificationException(PlatformErrorMessages.RPR_UNA_REGTYPE_NOT_MATCHING.getCode(),
+						String.format(PlatformErrorMessages.RPR_UNA_REGTYPE_NOT_MATCHING.getMessage(), rid));
+				}
+				
+				sendWorkflowCompletedWebSubEvent(registrationStatusDto);
+				regProcLogger.info("Request added to queue succesfully  for rid {}", rid);
+				description.setMessage(PlatformSuccessMessages.RPR_USER_NOTIFICATION_API_SUCCESS.getMessage());
+				description.setCode(PlatformSuccessMessages.RPR_USER_NOTIFICATION_API_SUCCESS.getCode());
+				updateAudit(description, rid, true, user);
+				buildResponse(ctx, description.getMessage(), null);
+				regProcLogger.debug("UserNotificationApi:processURL called ended for registration id {}", rid);
+
+		} catch (UserNotificationRequestValidationException e) {
+			logError(rid, e.getErrorCode(), e.getMessage(), e, ctx);
+		} catch (UserNotificationException e) {
+			logError(rid, e.getErrorCode(), e.getMessage(), e, ctx);
+		} catch (Exception e) {
+			logError(rid, PlatformErrorMessages.RPR_UNA_UNKNOWN_EXCEPTION.getCode(),
+					PlatformErrorMessages.RPR_UNA_UNKNOWN_EXCEPTION.getMessage(), e, ctx);
+		}
+
+	}
+
+	/**
+	 * @param routingContext
+	 */
+	private void failure(RoutingContext routingContext) {
+		this.setResponse(routingContext, routingContext.failure().getMessage());
+	}
+
+	/**
+	 *
+	 */
+	@Override
+	public MessageDTO process(MessageDTO object) {
+		return null;
+	}
+
+	/**
+	 * @param ctx
+	 * @return
+	 */
+	private String getUser(RoutingContext ctx) {
+		String user = "";
+		if (Objects.nonNull(ctx.user()) && Objects.nonNull(ctx.user().principal()))
+			user = ctx.user().principal().getString("username");
+		return user;
+	}
+
+	/**
+	 * Update audit.
+	 *
+	 * @param description             the description
+	 * @param registrationId          the registration id
+	 * @param isTransactionSuccessful the is transaction successful
+	 */
+	private void updateAudit(LogDescription description, String registrationId, boolean isTransactionSuccessful,
+			String user) {
+
+		String moduleId = isTransactionSuccessful ? MODULE_ID : description.getCode();
+
+		String eventId = isTransactionSuccessful ? EventId.RPR_402.toString() : EventId.RPR_405.toString();
+		String eventName = isTransactionSuccessful ? EventName.UPDATE.toString() : EventName.EXCEPTION.toString();
```

Comment:
In this audit for resent user notification, event name as udpate and related eventId is not making sense

Reply:
RPR_402 is Id for update event and RPR_405 is Id for exception event.